### PR TITLE
rust: 1.68.2 -> 1.69.0

### PR DESCRIPTION
**Pull request by nagisa**
Upgrade to this long awaited nice version that just so happened to have been released on a particularly well timed occassion.

### CODE REVIEW

**Comment by nagisa** [comment]
Oh, this will be all sorts of *fun*!

**Comment by nagisa** [comment]
The test failure is https://github.com/clap-rs/clap/issues/4857. The best way forward, I think, is to upgrade clap.

File: runtime/near-vm-runner/src/tests/cache.rs [hunk 1]
```diff
@@ -124,13 +124,13 @@ fn test_wasmer2_artifact_output_stability() {
     ];
     let mut got_prepared_hashes = Vec::with_capacity(seeds.len());
     let compiled_hashes = [
-        5254981150840481178,
-        15529260255496677612,
-        407257192602619216,
-        10913823971520273759,
-        17423008210698923502,
-        7011050181604188333,
-        15514788595649734538,
```
Comment (Ekleog-NEAR) [inline runtime/near-vm-runner/src/tests/cache.rs:133]: Hmm I’m curious, do you know why the hashes changed? I wouldn’t expect a rustc upgrade to change the hashes of wasmer binaries

Comment (nagisa) [reply runtime/near-vm-runner/src/tests/cache.rs:133]: `rkyv` is sensitive to `repr(Rust)` layout changes, unless the `strict` feature is enabled. Unfortunately that is something we cannot enable as we rely on being able to serialize tuples, and tuples are `repr(Rust)`. So something about how stuff is laid out there has changed.

**Comment by nagisa** [comment]
Blocked on https://github.com/near/nearcore/pull/8974

**Review by nikurt — APPROVED** [review]

**Pull request merged**